src/model/Document.cpp

```diff
@@ -407,13 +407,6 @@ auto Document::operator=(const Document& doc) -> Document& {
     buildContentsModel();
     updateIndexPageNumbers();
 
-    bool lastLock = tryLock();
```

Comment:
I guess it is assumed here that tryLock returns false if we already had the lock. But this is not the case. It's actually undefined.

GMutex is neither guaranteed to be recursive nor to be non-recursive, i.e. the return value of g_mutex_trylock() could be both FALSE or TRUE, if the current thread already has locked mutex. Use GStaticRecMutex, if you need recursive mutexes.
https://developer.gimp.org/api/2.0/glib/glib-Threads.html#g-mutex-trylock